File: 187. Repeated DNA Sequences/solution.go
func findRepeatedDnaSequences(s string) []string {
    seen := make(map[string]bool)
    repeat := make(map[string]bool)
    for i := 0; i + 10 <= len(s); i++ {
        if _, ok := seen[s[i:i+10]]; ok {
            repeat[s[i:i+10]] = true
        } else {
            seen[s[i:i+10]] = true
        }
    }
    
    res := make([]string, 0, len(repeat))
    for seq := range repeat {
        res = append(res, seq)
    }
    
    return res
}

File: 187. Repeated DNA Sequences/solution.py
class Solution:
    def findRepeatedDnaSequences(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        if len(s) < 10:
            return []

        seq = {}
        sol = []
        for i in range(len(s)):
            try:
                seq[s[i:i+10]] += 1
                if seq[s[i:i+10]] == 2:
                    sol.append(s[i:i+10])
            except:
                seq[s[i:i+10]] = 1
                
        return sol
